### DISCUSSION TITLE
Ansible and Kitty

### DISCUSSION THREAD

**Discussion by chriscroome** [kovidgoyal/kitty#5312] [General]
I've not found any issues or discussions about the use of Kitty and Ansible and would be interested if anyone has looked at all at how the [SSH kitten](https://sw.kovidgoyal.net/kitty/kittens/ssh/) and the [Ansible use of SSH](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/ssh_connection.html) interact and if there are any tweaks one can make to configurations to improve things, especially when using Yubikeys and SSH. Note that I haven't found any issues so far, everything appears to be working fine! Perhaps there is nothing that needs to be done!

I've been meaning to experiment with the [Ansible paramiko SSH](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/paramiko_ssh_connection.html) connection but haven't found time yet.

I found that the [Debian Bookworm](https://packages.debian.org/bookworm/kitty) version of Kitty, 0.21.2 was rather old so I wrote [an Ansible role to install the latest version of Kitty](https://github.com/webarch-coop/ansible-role-kitty) (initially in `/usr/local` until I realised that was a huge mistake :roll_eyes:) and then updated it to also install nightly and specific versions due to [this issue](https://github.com/kovidgoyal/kitty/issues/5188), I've now added it to [Ansible Galaxy](https://galaxy.ansible.com/chriscroome/kitty) in case anyone else wishes to install Kitty locally using Ansible.

**Comment by page-down** [comment]
> ... how the SSH kitten and the Ansible use of SSH interact ...
> ... experiment with the Ansible paramiko SSH connection ...

ssh kitten is for use in interactive scenarios. This feature relies on the kitty GUI.
You should call ssh directly when using ansible.
